response.go

```diff
@@ -207,6 +197,82 @@ func newBitmapResultFromInternal(bitmap *pbuf.Bitmap) (*BitmapResult, error) {
 	return result, nil
 }
 
+func (BitmapResult) Type() uint32                  { return QueryResultTypeBitmap }
+func (b BitmapResult) Bitmap() BitmapResult        { return b }
+func (BitmapResult) CountItems() []CountResultItem { return nil }
+func (BitmapResult) Count() int64                  { return 0 }
+func (BitmapResult) Sum() int64                    { return 0 }
+func (BitmapResult) Changed() bool                 { return false }
+
+func (b BitmapResult) MarshalJSON() ([]byte, error) {
+	bits := b.Bits
+	if bits == nil {
+		bits = []uint64{}
+	}
+	keys := b.Keys
+	if keys == nil {
+		keys = []string{}
+	}
+	return json.Marshal(struct {
+		Attributes map[string]interface{} `json:"attrs"`
+		Bits       []uint64               `json:"bits"`
+		Keys       []string               `json:"keys"`
+	}{
+		Attributes: b.Attributes,
+		Bits:       bits,
+		Keys:       keys,
+	})
+}
+
+type SumCountResult struct {
+	sum   int64
+	count int64
+}
```

Comment:
Is it possible to change this to
```go
    Sum   int64 `json:"sum"`
    Count int64 `json:"count"`
```

rather than having the separate MarshalJSON method? Or is there somethi

Reply:
That's spot on. Updated.

Reply:
I just remembered. `QueryResult` interface has both `Sum()` and `Count()` functions, so it's not possible to have `Sum`, `Count` fields in `SumCountResult`. That requires adding `MarshalJSON` function to that struct.

Reply:
Oh I see - I would prefer if we just changed the names slightly rather than having the separate MarshalJSON. SumVal and CountVal maybe? then just leave the json tags the same.

Reply:
ok